### PR TITLE
feat(node): v18

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V18.md#18.0.0)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### Current state

- Fixes my migration scripts (oops)
- Rolls over types ONLY

We should probably keep this PR open and aim any v18 related changes into here until we have at least all breaking changes collected.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
@SimonSchick so this one looks perfectly fine, a technical one. What next step?

**Comment by SimonSchick** [comment]
We can either merge and release this and intro breaking changes later or keep this PR open (as described) until we have all of the major breaking changes merged in.

The former will cause more churn for developers, the latter might lead to many people filings PRs against v17 that are incorrect.

**Comment by peterblazejewicz** [comment]
this is why I asked. How to handle the PR like that one?
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60042
It looks like breaking change under v18, it cannot be accepted under current version.

**Comment by SimonSchick** [comment]
I can cherry pick these commits into this branch and we can close the other PR, you *should* have push permissions to my fork so it should work by just changing the merge target branch.

**Comment by glasser** [comment]
Does this PR include the addition of `fetch` to the global namespace?

**Comment by SimonSchick** [comment]
No.

**Comment by SimonSchick** [comment]
I will perform a final rebase to ensure v17 types are correct before this is merged once approved, there's likely more changes but I currently do not have the time to go over all of them.

**Comment by glasser** [comment]
FWIW it looks like the `family` change will be reverted: https://github.com/nodejs/node/pull/43054

**Comment by 1valdis** [comment]
>> Does this PR include the addition of fetch to the global namespace?

> No

@SimonSchick could you explain why? I'd expect it be in the global namespace, given that it's now enabled by default.

**Comment by SimonSchick** [comment]
Can be added later (as non breaking change), takes a lot of work. I need to get this PR in some time this week.

**Comment by SimonSchick** [comment]
Rebased everything, will have to make sure all tests pass, we should move ahead with this asap, sorry for the delay, been busy.

**Comment by SimonSchick** [comment]
Test failure is unrelated to PR, requesting review cc @peterblazejewicz

**Comment by peterblazejewicz** [comment]
I saw this, looked into DT module, it's an old one, not worth to rewrite, I'll remove the block that was not failing prior 4.8 (as per inline comments)

**Comment by peterblazejewicz** [comment]
#60804 60804

**Comment by SimonSchick** [comment]
*sigh*

File: types/node/url.d.ts
Comment (yoursunny) [inline types/node/url.d.ts:69]: You have two "the"s at the end of this line.

Comment (SimonSchick) [reply types/node/url.d.ts:69]: I did not write this, out of scope.

Comment (peterblazejewicz) [reply types/node/url.d.ts:69]: fyi: came from master changes

**Comment by peterblazejewicz** [comment]
@SimonSchick I beieve it was already reviewed and it's ready to go. Are you positive on this? We could make it happen

**Comment by SimonSchick** [comment]
Should be ok now, this may not reflect the full extend to v18 but this is all I have time for atm.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@SimonSchick thanks a ton!

I've edited the title to reflect non-wip

**Review by yoursunny — APPROVED** [review]
Tested in [NDNts](https://github.com/yoursunny/NDNts) codebase and no error found.

**Comment by SimonSchick** [comment]
Ready to merge

**Pull request merged**
